internal/provider/models/alerts/base_model.go

```diff
@@ -375,15 +380,59 @@ func GetAlertPayloadFromModel(v attr.Value, dd *diag.Diagnostics) map[string]any
 		"name": serviceName,
 	}
 
-	if serviceName != "log_analysis" && serviceValues["uri"].(StringValue).IsNull() {
+	// Check that there are no extra properties that the service doesn't expect
+	for attribute := range serviceValues {
+		if attribute == "name" {
+			continue
+		}
+		switch attribute {
+		case "uri":
+			if serviceName == "slack" || serviceName == "pager_duty" || serviceName == "webhook" {
+				continue
+			}
+		case "message_text":
+			if serviceName == "slack" || serviceName == "webhook" {
+				continue
+			}
+		case "summary", "source", "routing_key", "event_action":
+			if serviceName == "pager_duty" {
+				continue
+			}
+		case "severity":
+			if serviceName == "pager_duty" || serviceName == "log_analysis" {
+				continue
+			}
+		case "subject", "body", "ingestion_key":
+			if serviceName == "log_analysis" {
+				continue
+			}
+		case "auth", "headers", "method":
+			if serviceName == "webhook" {
+				continue
+			}
+		}
+
+		if serviceValues[attribute].IsNull() {
+			// The TF framework will populate optional fields with null values. Values must
+			// be non-null to be considered an error
+			continue
+		}
 		dd.AddError(
 			"Error in plan",
-			"`uri` is required for Slack, PagerDuty, or Webhook",
+			fmt.Sprintf("Attribute `%s` is not allowed for service `%s`", attribute, serviceName),
 		)
 	}
 
+	// Check for *missing* properties for the selected service, and construct the payload.
+	// At this point, there are no extra properties which is validated above.
 	switch serviceName {
 	case "slack", "webhook":
+		if serviceValues["uri"].(StringValue).IsNull() {
+			dd.AddError(
+				"Error in plan",
+				"`uri` is required for Slack, PagerDuty, or Webhook",
```

Comment:
This error includes PD but the `case` does not

Reply:
Yep, thanks, this could be tuned.  This error used to be at the top but was re-worked once I reworked the error checking to error for all missing fields as well as extra (unexpected) fields.

This error now appears in 2 of the case statements, so I'll narrow the scope of the message.

Reply:
fixed